### ISSUE TITLE
Object does not exists in mock database after test

### ISSUE THREAD

**Issue by paveltsytovich** [ioBroker/testing#48]
Hi,

I want to check my code for correct remove object from iobroker database. I have following preset objects:
```
predefinedObjects: [
		{
			'_id': 'noolitef.0.testlamp', 
			'type': 'channel',
			// @ts-ignore
			'parent': 'noolitef.0',        
			 'children': [
				'noolitef.0.testlamp.status',
				'noolitef.0.testlamp.channel'
			 ],
			'common': {
				'name':  'testlamp',      
				'role':  'light.switch',         
				'desc':  'fill comment for test purpose'                     
			}
		 },
		 {
			'_id': 'noolitef.0.badlamp', 
			'type': 'channel',
			// @ts-ignore
			'parent': 'noolitef.0',         
			 'children': [
				'noolitef.0.badlamp.status',
				'noolitef.0.baslamp.channel'
			 ],
			'common': {
				'name':  'badlamp',      
				'role':  'light.switch',          
				'desc':  'fill comment for test purpose'                   
			}
		 }
	],
	predefinedStates: {
		'noolitef.0.testlamp.status' : {'val' :false, 'ack':false},
		'noolitef.0.testlamp.channel' : {'val' :1, 'ack' : false}
	}
```

In main.js I do nothing and then I try to check exits object in database:

`const result = database.hasObject('noolitef.0.badlamp');`

The _'result'_ is false. I think that object _'noolitef.0.badlamp'_ must be exists and _'result'_ must be 'true'

**Comment by AlCalzone** [comment]
I'll check it out. Can you share a permalink to your repo where this fails?

**Comment by paveltsytovich** [comment]
https://github.com/paveltsytovich/ioBroker.noolitef/blob/feature/Binding/test/unit.js 
and branch https://github.com/paveltsytovich/ioBroker.noolitef/tree/feature/Binding

**Comment by AlCalzone** [comment]
For the tests you define inside `defineAdditionalTests`, you need to add your own `before[Each]` handler to populate the DBs, see documentation:
> ```js
>     // Define your own tests inside defineAdditionalTests. 
>     // If you need predefined objects etc. here, you need to take care of it yourself
>     defineAdditionalTests() {
>         // ...
>     },
> ```

But I think that is not what you want. You want to perform the adapter startup and **then** assert certain things, correct?
I would need to add another hook to the startup test then. Before I do, I want to make sure that is what you need.

**Comment by paveltsytovich** [comment]
Yes. I want sync object from config with iobroker database during adapter is starting. As admin I can remove some object or add in config via index_m.html page, and then I want remove  or add objects in iobroker database accoring my config
